src/main/java/bio/terra/common/SynapseColumn.java

```diff
@@ -118,15 +134,7 @@ static boolean checkForJSONCastRequirement(TableDataType dataType, boolean isArr
     if (isArrayOf) {
       return false;
     }
-    switch (dataType) {
-      case DIRREF:
-      case FILEREF:
-      case TEXT:
-      case STRING:
-        return false;
-      default:
-        return true;
-    }
+    return dataType.equals(TableDataType.TEXT);
```

Comment:
Why is the TEXT data type handled differently?

Reply:
Ah, that was left over from testing.  I think I got rid of this logic

Reply:
I think it's still there: https://github.com/DataBiosphere/jade-data-repo/pull/1568/files#diff-4b4d66994f401ea0167fb07c4dbc16d7940baba87260e1c427e849eb74e864e7R136